concepts/pivoted/eICU-pivoted-OASIS.sql

```diff
@@ -0,0 +1,571 @@
+    WHEN vent_4 = 1 THEN 9
+    WHEN vent_5 = 1 THEN 9
+    WHEN vent_6 = 1 THEN 9
+    ELSE 0
+    END AS VENT_OASIS
+    FROM merged_vent
+    --WHERE (chartoffset > 0 AND chartoffset <= 1440) -- already considered in step above
+    GROUP BY pid_vent, vent_1, vent_2, vent_3, vent_4, vent_5, vent_6
+    -- possible: error - only 8886 cases from ventilation_events table but not more
+    -- error in derived.vent_events? no reference/code on github
+)
+
+, cohort_OASIS AS (
+  SELECT cohort.patientunitstayid, 
+  pre_ICU_LOS_OASIS.pre_ICU_LOS_OASIS, 
+  age_OASIS.age_OASIS, 
+  GCS_OASIS.GCS_OASIS,
+  heartrate_OASIS.heartrate_OASIS,
+  MAP_OASIS.MAP_OASIS,
+  respiratoryrate_OASIS.respiratoryrate_OASIS,
+  temperature_OASIS.temperature_OASIS,
+  urineoutput_OASIS.urineoutput_OASIS,
+  VENT_OASIS.VENT_OASIS,
+  electivesurgery_OASIS.electivesurgery_OASIS
+  FROM  `physionet-data.eicu_crd.patient` AS cohort
+
+  LEFT JOIN pre_ICU_LOS_OASIS
+  ON cohort.patientunitstayid = pre_ICU_LOS_OASIS.pid_LOS
+
+  LEFT JOIN age_OASIS
+  ON cohort.patientunitstayid = age_OASIS.pid_age 
+
+  LEFT JOIN GCS_OASIS
+  ON cohort.patientunitstayid = GCS_OASIS.pid_GCS 
+
+  LEFT JOIN heartrate_OASIS
+  ON cohort.patientunitstayid = heartrate_OASIS.pid_HR 
+
+  LEFT JOIN MAP_OASIS
+  ON cohort.patientunitstayid = MAP_OASIS.pid_MAP 
+
+  LEFT JOIN respiratoryrate_OASIS
+  ON cohort.patientunitstayid = respiratoryrate_OASIS.pid_RR
+
+  LEFT JOIN temperature_OASIS
+  ON cohort.patientunitstayid = temperature_OASIS.pid_temp
+
+  LEFT JOIN urineoutput_OASIS
+  ON cohort.patientunitstayid = urineoutput_OASIS.pid_urine
+
+  LEFT JOIN VENT_OASIS
+  ON cohort.patientunitstayid = VENT_OASIS.pid_vent
+
+  LEFT JOIN electivesurgery_OASIS
+  ON cohort.patientunitstayid = electivesurgery_OASIS.pid_adm
+
+)
+
+, score_impute AS (
+
+SELECT cohort_OASIS.*,
+  IFNULL(pre_ICU_LOS_OASIS, 5) AS pre_ICU_LOS_OASIS_W,
+  IFNULL(age_OASIS, 9) AS age_OASIS_W, 
+  IFNULL(GCS_OASIS, 10) AS GCS_OASIS_W, 
+  IFNULL(heartrate_OASIS, 6) AS heartrate_OASIS_W,
+  IFNULL(MAP_OASIS, 4) AS MAP_OASIS_W,
+  IFNULL(respiratoryrate_OASIS, 10) AS respiratoryrate_OASIS_W,
+  IFNULL(temperature_OASIS, 6) AS temperature_OASIS_W, 
+  IFNULL(urineoutput_OASIS, 10) AS urineoutput_OASIS_W, 
+  IFNULL(VENT_OASIS, 9) AS VENT_OASIS_W, 
+  IFNULL(electivesurgery_OASIS, 6) AS electivesurgery_OASIS_W, 
+  
+  IFNULL(pre_ICU_LOS_OASIS, 0) AS pre_ICU_LOS_OASIS_B,
+  IFNULL(age_OASIS, 0) AS age_OASIS_B, 
+  IFNULL(GCS_OASIS, 0) AS GCS_OASIS_B, 
+  IFNULL(heartrate_OASIS, 0) AS heartrate_OASIS_B,
+  IFNULL(MAP_OASIS, 0) AS MAP_OASIS_B,
+  IFNULL(respiratoryrate_OASIS, 0) AS respiratoryrate_OASIS_B,
+  IFNULL(temperature_OASIS, 0) AS temperature_OASIS_B, 
+  IFNULL(urineoutput_OASIS, 0) AS urineoutput_OASIS_B, 
+  IFNULL(VENT_OASIS, 0) AS VENT_OASIS_B, 
+  IFNULL(electivesurgery_OASIS, 0) AS electivesurgery_OASIS_B, 
+
+FROM cohort_OASIS
+)
+
+--Compute overall scores -> No imputation, then Worst, then Best Case Scenario
```

Comment:
Interesting approach! It's a bit non-standard though. I might suggest: retain the original values (with NULL), and calculate the scores using only the best case scenario (which is the accepted approach for severity of illness scores). So you would have MAP_OASIS (with NULL values where data is missing), MAP_OASIS_score (imputing 0 for NULL values), and then the final OASIS_score and OASIS_prob.

Reply:
We didn't know that it is common practice to always assume a "best case" with scores why we have calculated both possibilities. To keep it simple, I deleted the "worst case", but kept the rest. This way users can decide what they want to use. Hope that's okay?

Reply:
I think it's less assuming best case and more acknowledging that care providers have a bias to only measure physiology when they suspect it is abnormal. In that sense, the lack of measurement is an implication that the measurement is normal, so we assume a score of 0.